### DISCUSSION TITLE
Set DOM's x-text to a returned object's key:value

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3594] [1. Help]
```
One-Time Cost : $<span id="one-time" x-text="({ monthly: obj.monthly, onetime: obj.onetime } = await foo())"></span>
```

How do I get **one-time**'s `x-text` to be `obj.onetime` ?

**Comment by ekwoka** [comment]
just return only that.

But also, what you wrote there isn't valid js anyway

```html
x-text="(await foo()).onetime"
```

**Reply by anjanesh** [reply]
```
x-text="({ monthly: obj.monthly, onetime: obj.onetime } = await foo()).onetime"
```

**Reply by ekwoka** [reply]
No...that's still invalid JS. `Illegal property in declaration context`

I gave you the correct way to do it.

**Reply by anjanesh** [reply]
I myself don't understand it - really - but this was ChatGPT's answer.
https://javascript.co.in/something-totally-new-i-learnt-in-javascript-from-chat-gpt
And it works in my Chrome and FireFox.

**Reply by ekwoka** [reply]
oh, I didn't really notice the fact there isn't a `const` and that makes a difference..

Wow, interesting.

But, still a bad way to do this.

```
(await foo()).onetime
```

That's the way
